src/vstest.console/InProcessVsTestConsoleWrapper.cs

```diff
@@ -133,6 +136,30 @@ public InProcessVsTestConsoleWrapper(ConsoleParameters consoleParameters)
 
     internal ITestRequestManager? TestRequestManager { get; set; }
 
+    private void UpdateDotnetLookupPath()
+    {
+        const string dotnetRootOverrideEnvVarName = "VSTEST_WINAPPHOST_DOTNET_ROOT";
+        const string programFilesPathEnvVarName = "ProgramFiles";
+        const string dotnetGlobalInstallationDir = "dotnet";
+
+        // Allow users to specify a dotnet private installation path before attempting to use the
+        // global installation path.
+        if (_environmentVariableHelper.GetEnvironmentVariable(dotnetRootOverrideEnvVarName) is not null)
+        {
+            return;
+        }
+
+        var dotnetPathPrefix = _environmentVariableHelper.GetEnvironmentVariable(programFilesPathEnvVarName);
```

Comment:
@MarcoRossignoli Can you think of a better way to retrieve the "default" global installation folder?

Reply:
In windows can be in different way...99.9% I expect to find it there...but it's possible that user will install in different folder.
The "correct code" is here:

https://github.com/microsoft/vstest/blob/main/src/Microsoft.TestPlatform.CoreUtilities/Helpers/DotnetHostHelper.cs#L121

Reply:
Closing this as we abandoned the idea of patching the dotnet location.